### PR TITLE
Add cJSON Zephyr Module

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  remotes:
+    - name: mender
+      url-base: https://github.com/mendersoftware
+
+  projects:
+    - name: cjson-zephyr
+      submodules: true
+      remote: mender
+      path: modules/lib/cjson-zephyr
```
Comment (vpodzime) [inline west.yml:11]: Is this where it goes when fetching/checking out locally?

**Review by vpodzime — APPROVED** [review]
Looks good to me.

File: zephyr/Kconfig [hunk 2]
```diff
@@ -424,4 +424,21 @@ if MENDER_MCU_CLIENT
 
     endif
 
+    menu "Optional functionality (ADVANCED)"
+
+        config MENDER_FULL_PARSE_ARTIFACT
+            bool "Full Mender Artifact parse"
+            default y
+            help
+                Do full parsing of the Artifacts allowoing for robustness feature of matching the device type
```
Comment (larsewi) [inline zephyr/Kconfig:433]:
```suggestion
                Do full parsing of the Artifacts allowing for robustness feature of matching the device type
```

**Review by larsewi — APPROVED** [review]

File: west.yml (hunk 1 above)
Comment (lluiscampos) [reply west.yml:11]: Yes!

It is a bit arbitrary, but in `modules/lib` I found similar libraries like `zcbor`

Comment (lluiscampos) [reply west.yml:11]: But actually, I made a mistake in the final submit: I wanted it to be `modules/lib/cjson` only, without the `-zephyr` suffix which in this context is superfluous. I'll amend this and merge :+1:

File: zephyr/Kconfig (hunk 2 above)
Comment (lluiscampos) [reply zephyr/Kconfig:433]: :pray:

**Pull request merged**